kuma/users/views.py

```diff
@@ -599,31 +599,31 @@ def get_form(self, form_class=None):
         form = super(SignupView, self).get_form(form_class)
         form.fields["email"].label = _("Email address")
 
-        User = get_user_model()
-
-        # When no username is provided, default to the local-part of the email address
-        if form.initial.get("username", "") == "":
-            email = form.initial.get("email", "")
-            if isinstance(email, tuple):
-                email = email[0]
-            suggested_username = suggested_username_base = email.split("@")[0]
+        # We should only see GitHub/Google users.
+        assert self.sociallogin.account.provider in ("github", "google")
+
+        initial_username = form.initial.get("username")
```

Comment:
Down on line 624 there's that strange...
```python
if initial_username:
    ...
else:
    initial_username = ""
```
I think a more idiomatic syntax would be to use 
```suggestion
        initial_username = form.initial.get("username") or ""
```

Reply:
Yep, that's better @peterbe, thanks. I think instead but almost the same I'll use:
```python
initial_username = form.initial.get("username", "")
```

Reply:
But then `intial_username` might become `None` something else falsy. 
But storm in a teacup, I don't think it's a risk in the current context.

Reply:
Ah, yes, I suppose the actual value of the `username` key could be `None` so yours is safer. Sorry, I'll adjust that in my clean-up PR.